src/main/java/ch/njol/skript/expressions/ExprHatchingNumber.java

```diff
@@ -0,0 +1,105 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerEggThrowEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Hatching Number")
+@Description({
+	"The number of entities that will be hatched in a Player Egg Throw event.",
+	"Please note that no more than 127 entities can be hatched at once."
+})
+@Events({
+	"on player egg throw:",
+	"\tset the hatching number to 10"
+})
+@Since("INSERT VERSION")
+public class ExprHatchingNumber extends SimpleExpression<Byte> {
+
+	static {
+		Skript.registerExpression(ExprHatchingNumber.class, Byte.class, ExpressionType.SIMPLE,
+			"[the] hatching number",
+			"[the] number of entities to hatch"
```

Comment:
Syntax sounds weird here, it may conflict with ExprAmount, not sure what it would be tho, can't think of a solution for now